scripts/chef-automate-install.sh

```diff
@@ -0,0 +1,213 @@
+#/ Options:
+#/   --help: Display this help message
+#/   --debug: Triggers trace output on the bash script to help with troubleshooting
+#/   --app-id: Application ID
+#/   --tenant-it: Tenant ID
+#/   --password: Password
+#/   --key-vault-name: Name of the aure key vault that stores all the secrets
+usage() { grep '^#/' "$0" | cut -c4- ; exit 0 ; }
+
+# initialize variables
+appID=""
+tenantID=""
+password=""
+objectId=""
+keyVaultName=""
+PARAMS=""
+while (( "$#" )); do
+  case "$1" in
+    -d|--debug)
+      set -o xtrace
+      shift 1
+      ;;
+    -h|--help)
+      usage
+      ;;
+    -a|--app-id)
+      appID=$2
+      shift 2
+      ;;
+    -t|--tenant-id)
+      tenantID=$2
+      shift 2
+      ;;
+    -p|--password)
+      password=$2
+      shift 2
+      ;;
+    -k|--key-vault-name)
+      keyVaultName=$2
+      shift 2
+      ;;
+    -o|--object-id)
+      objectId=$2
+      shift 2
+      ;;
+    --) # end argument parsing
+      shift
+      break
+      ;;
+    -*|--*=) # unsupported flags
+      echo "Error: Unsupported flag $1" >&2
+      exit 1
+      ;;
+    *) # preserve positional arguments
+      PARAMS="$PARAMS $1"
+      shift
+      ;;
+  esac
+done
+
+# set positional arguments in their proper place
+eval set -- "$PARAMS"
+
+# Set magic variables for current file & dir
+__dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+__file="${__dir}/$(basename "${BASH_SOURCE[0]}")"
+__base="$(basename "${__file}" .sh)"
+__root="$(cd "$(dirname "${__dir}")" && pwd)" # <-- change this as it depends on your app
+
+# Setup logging
+readonly LOG_FILE="/tmp/$(basename "$0").log"
+readonly DATE_FORMAT="+%Y-%m-%d_%H:%M:%S.%2N"
+info()    { echo "[$(date ${DATE_FORMAT})] [INFO]    $*" | tee -a "$LOG_FILE" >&2 ; }
+warning() { echo "[$(date ${DATE_FORMAT})] [WARNING] $*" | tee -a "$LOG_FILE" >&2 ; }
+error()   { echo "[$(date ${DATE_FORMAT})] [ERROR]   $*" | tee -a "$LOG_FILE" >&2 ; }
+fatal()   { echo "[$(date ${DATE_FORMAT})] [FATAL]   $*" | tee -a "$LOG_FILE" >&2 ; exit 1 ; }
+
+# --- Helper scripts end ---
+
+_downloadSecretsFromAzureKeyVault() {
+	apt-get install -y libssl-dev libffi-dev python-dev build-essential
+	AZ_REPO=$(lsb_release -cs)
+	echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ ${AZ_REPO} main" | sudo tee /etc/apt/sources.list.d/azure-cli.list
+	curl -L https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+	apt-get update
+	apt-get install -y azure-cli
+	apt-get install -y apt-transport-https
+	apt-get update
+	apt-get install -y sshpass
+	apt-get install -y wget
+	apt-get install -y lvm2 xfsprogs sysstat atop
+
+	az login --service-principal -u "${appID}" --password "${password}" --tenant "${tenantID}"
+	az keyvault secret download --file "${DELIVERY_DIR}/chefautomatedeliveryuser.pem" --name chefdeliveryuserkey --vault-name "${keyVaultName}"
+    	return
+}
+
+: '
+NOTE: Not sure that we need this anymore.
```

Comment:
Would be good to get clarity on this either way and remove/add back to the main method accordingly.